### ISSUE TITLE
Add a logger

### ISSUE THREAD

**Issue by felipedau** [AnemoneLabs/unmessage#30]
It is currently not very simple to debug unMessage (specially when using the CLI) and having a logger available would be a great improvement.

**Comment by felipedau** [comment]
Hi @david415 and @meejah, I took a look at Twisted's logger for this task as you guys suggested and wrote one for unMessage - it's in the [feature/logger](https://github.com/felipedau/unmessage/tree/feature/logger) branch .

What I had in mind was to save separate logs for each peer in their respective dirs and for that reason I used the `globalLogBegginer` to forward logs to the observer once the peer is created (it's basically after the user provides a name so it can create the dir and files). I think that due to the global log, logs from Twisted and txtorcon were also captured, so I had to create a pretty bad filter checking the namespace starts with `'unmessage'` :/ it still does not filter out messages like this:

```
[unmessage.peer._ConversationFactory#info] Stopping factory <unmessage.peer._ConversationFactory instance at 0x7f77a3be0bd8>
```

I am not sure if there is a nice way to do that than hardcoding a list of the exact namespaces I would like to filter. Do you have any suggestions?

I also added a filter for the log level because I wanted to let users customize that (but right now the interfaces are forcing `debug`).

The `textFileLogObserver` is great - it works very well and has a nice formatting. Unfortunately for the `STDLibLogObserver`, it uses the formatting of the default stdlib logger which IMO is not as good as the text file observer's. As a test, I created a simple `PrintObserver` to use the `formatEvent` of `textFileLogObserver`, but it failed to log when using the GUI, probably because it is not thread safe. I switched back to the stdlib logger for now so maybe I have to find a way to either use the text file `formatEvent` or just format it myself.

In `__main__` I did not pass the `gui` module in the event because I am not sure if passing a whole module is a good idea, just the name. Is it?

For this test I thought it would be a good idea to log the current bootstrap process as that is a very important part and it works.

I only have some concerns about storing information of a privacy/anonymity enhancing app on disk, so maybe the default level in the future should be `warn` and do not store logs by default, unless the users says so? Once we start passing "attr'ed" objects to the events, we might have to make filters to sanitize that info too, right?

Let me know what you think!

**Comment by felipedau** [comment]
The logger has been working for a while and what was left was just adding some log statements to important parts of the code. The merge commit 1e1d58c3350306c0780e828aac7f337cef051f0c into develop contains the last changes to finish this task.

While I was adding the statements, I also made some improvements:

- Fix a bug related to the file logger - it was being started before a new peer's directory was created (0615dbffe282b80c5876bb88a5baf1099584ee3d)
- Filter specific namespaces but allow any event with levels higher than `LogLevel.warn`
- Use the `loggerFor` factory for more complete namespaces, which also works nicely with attrs (f3e83bc77ad2bea36a3fa8eedd71da33157ea593)

As @david415 and I had already discussed the status of this task and only these small changes were left to be done, I am closing this.

Thanks!
